src/main/twirl/gitbucket/core/helper/diff.scala.html

```diff
@@ -101,34 +103,112 @@
             <div class="diff-same">File mode changed</div>
           }
         } else {
-          @if(diff.newContent != None || diff.oldContent != None){
-            <div id="diffText-@i" class="diffText"></div>
-            <textarea id="newText-@i" style="display: none;" data-file-name="@diff.oldPath" data-val='@diff.newContent.getOrElse("")'></textarea>
-            <textarea id="oldText-@i" style="display: none;" data-file-name="@diff.newPath" data-val='@diff.oldContent.getOrElse("")'></textarea>
-          } else {
-            @if(diff.newIsImage || diff.oldIsImage){
-              <div class="diff-image-render diff2up">
-              @if(oldCommitId.isDefined && diff.oldIsImage){
-                <div class="diff-image-frame diff-old"><img src="@helpers.url(repository)/raw/@oldCommitId.get/@diff.oldPath" class="diff-image" onload="onLoadedDiffImages(this)" style="display:none" /></div>
-              } else {
-                @if(diff.changeType != ChangeType.ADD){
-                  <div style="padding: 12px;">Not supported</div>
+          @(diff.oldContentInfo, diff.newContentInfo) match {
+            case (None, None) => {
+              <div>Error???</div>
+            }
+            case (Some(oldContentInfo: ContentInfo), Some(newContentInfo: ContentInfo)) if oldContentInfo.renderer.isInstanceOf[DiffRenderer] && newContentInfo.renderer.isInstanceOf[DiffRenderer] => {
```

Comment:
I wonder if renderer can be extracted from `ContentInfo` using pattern matching, not type casting. See also above commet for `ContentInfo`.